mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -154,8 +153,12 @@ def download_artifacts(self, artifact_path, dst_path=None):
             return local_dir
 
     def _download_file(self, remote_file_path, local_path):
-        raise MlflowException(
```

Comment:
Why don't you create the hdfs_artifact_repo object + use download_artifact ?. It will manage dir also.

Reply:
discard. (IRL discussion)